xscen/aggregate.py

```diff
@@ -137,8 +137,8 @@ def climatological_mean(
             )
         elif isinstance(ds.indexes["time"], xr.coding.cftimeindex.CFTimeIndex):
             time_coord = [
-                cftime.datetime(
-                    y - window + 1, m, d, calendar=ds.indexes["time"].calendar
+                xclim.core.calendar.datetime_classes[ds.indexes["time"].calendar](
```

Comment:
Not really a suggestion, but in case you didn't know, this also exists in xarray:
```suggestion
                xclim.core.calendar.datetime_classes[ds.dt.calendar](
```
It does pretty much the same as your line though,

Reply:
It does look cleaner!